File: backend/app/models/agent_file.py
import uuid
from datetime import datetime, timezone
from sqlalchemy import String, Integer, DateTime, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from app.core.database import Base


class AgentFile(Base):
    __tablename__ = "agent_files"

    id: Mapped[str] = mapped_column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    agent_id: Mapped[str | None] = mapped_column(
        String(36), ForeignKey("agents.id", ondelete="SET NULL"), nullable=True, index=True
    )
    user_id: Mapped[str] = mapped_column(
        String(36), ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True
    )
    filename: Mapped[str] = mapped_column(String(255), nullable=False)
    file_type: Mapped[str | None] = mapped_column(String(50), nullable=True)
    file_size: Mapped[int | None] = mapped_column(Integer, nullable=True)
    storage_path: Mapped[str | None] = mapped_column(String(500), nullable=True)
    description: Mapped[str | None] = mapped_column(String(500), nullable=True)
    created_by: Mapped[str] = mapped_column(String(20), default="agent")
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=lambda: datetime.now(timezone.utc)
    )

File: ios/Clawzy/clawzy/Models/AgentFile.swift
import Foundation

struct AgentFile: Codable, Identifiable {
    let id: String
    let agentId: String?
    let userId: String
    let filename: String
    let fileType: String?
    let fileSize: Int?
    let storagePath: String?
    let description: String?
    let createdBy: String
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, filename, description
        case agentId = "agent_id"
        case userId = "user_id"
        case fileType = "file_type"
        case fileSize = "file_size"
        case storagePath = "storage_path"
        case createdBy = "created_by"
        case createdAt = "created_at"
    }
}
